braidpool-primitives/src/braid/mod.rs

```diff
@@ -85,6 +98,38 @@ impl Braid {
 }
 
 impl Braid {
+    // All pub(crate) functions go here!
+    #[inline]
+    pub(crate) fn load_bead_from_hash(&self, bead_hash: BeadHash) -> Result<&Bead, BeadLoadError> {
+        // This functions returns a bead from memory! Future DB work goes in here!
+
+        // TODO: Add in a check for whether a bead_hash is valid!
+
+        match self.loaded_beads_in_memory.get(&bead_hash) {
+            Some(bead) => Ok(bead),
+            None => Err(BeadLoadError::BeadPruned),
+        }
+    }
+}
+
+impl Braid {
+    fn update_parents_of_bead(&self, child_bead: &Bead) {
```

Comment:
The parents of a bead are static, and committed to when the bead is created. Setting the parents should be part of a constructor. We should never be adding parents.

If e.g. we're mining on a bead having parents A and B, and a new bead C comes in, we should delete the previous bead we were mining on and instantiate a new Bead object with A,B,C. Adding the parent C changes the coinbase tx which commits to parents in an OP_RETURN (committed metadata), which changes the merkle root, and changes the bead entirely. There's no need to keep the old (A,B parent) bead around, it will never be mined and can be discarded.